uisdk/src/main/java/com/karhoo/uisdk/screen/booking/checkout/component/views/CheckoutViewPresenter.kt

```diff
@@ -191,6 +191,10 @@ internal class CheckoutViewPresenter(
     override fun consumeBackPressed(): Boolean {
         return if (view?.isPassengerDetailsViewVisible() == true) {
             view?.showPassengerDetailsLayout(false)
+            view?.forceValidation()
+            if(view?.arePassengerDetailsValid() == false){
```

Comment:
Why clear only if they aren't valid. This task may require us to rethink the screen a bit. The expected behaviour is that when the user presses back any modifications made from the initial state is discarded. So, for example, if the form is blank, the user fills in all or partial details and presses back, everything is discarded. But if there is passenger data when entering the screen and the user modifies something then presses back, then the initial passenger details remain intact and only the modifications are discarded.